File: Java/Maths/PowerSet.java
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Scanner;
import java.util.Set;

public class PowerSet {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the number of elements in the original set: ");
        int numElements = scanner.nextInt();

        List<Character> originalSet = new ArrayList<>();
        System.out.println("Enter the elements of the original set: ");
        for (int i = 0; i < numElements; i++) {
            originalSet.add(scanner.next().charAt(0));
        }

        // Remove duplicates from the original set
        Set<Character> uniqueSet = new HashSet<>(originalSet);
        originalSet.clear();
        originalSet.addAll(uniqueSet);

        // Generate the power set
        int n = originalSet.size();
        int numSubsets = (int) Math.pow(2, n);
        List<List<Character>> powerSet = new ArrayList<>();

        for (int i = 0; i < numSubsets; i++) {
            List<Character> subset = new ArrayList<>();
            for (int j = 0; j < n; j++) {
                if ((i & (1 << j)) != 0) {
                    subset.add(originalSet.get(j));
                }
            }
            powerSet.add(subset);
        }

        // Print the power set
        System.out.println("Power Set:");
        for (List<Character> subset : powerSet) {
            System.out.print("{ ");
            for (Character element : subset) {
                System.out.print(element + " ");
            }
            System.out.println("}");
        }
    }
}

File: Python/Maths/Power_Set.py
import itertools

def generate_power_set(original_set):
    n = len(original_set)
    power_set = []

    for i in range(2**n):
        subset = [original_set[j] for j in range(n) if (i & (1 << j)) != 0]
        power_set.append(subset)

    return power_set

def print_power_set(power_set):
    for subset in power_set:
        print("{", end=" ")
        print(*subset, end=" ")
        print("}")

def remove_duplicates(original_set):
    return list(set(original_set))

num_elements = int(input("Enter the number of elements in the original set: "))
original_set = []
print("Enter the elements of the original set: ")
for _ in range(num_elements):
    original_set.append(input())

# Remove duplicates from the original set
original_set = remove_duplicates(original_set)

# Generate the power set
power_set = generate_power_set(original_set)

# Print the power set
print("Power Set:")
print_power_set(power_set)